runtime/compiler/control/J9CompilationStrategy.hpp

```diff
@@ -20,24 +20,32 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
-#ifndef COMPILATIONCONTROLLER_INCL
-#define COMPILATIONCONTROLLER_INCL
+#ifndef J9_COMPILATIONSTRATEGY_INCL
+#define J9_COMPILATIONSTRATEGY_INCL
 
+/*
+ * The following #define and typedef must appear before any #includes in this file
+ */
+#ifndef J9_COMPILATIONSTRATEGY_CONNECTOR
+#define J9_COMPILATIONSTRATEGY_CONNECTOR
+   namespace J9 { class CompilationStrategy; }
+   namespace J9 { typedef J9::CompilationStrategy CompilationStrategyConnector; }
+#endif
+
+#include "control/OMRCompilationStrategy.hpp"
 #include "compile/CompilationTypes.hpp"
-#include "control/OptimizationPlan.hpp"
 #include "runtime/CodeCacheManager.hpp"
-#include "env/VMJ9.h"
 
-class TR_OptimizationPlan;
-class TR_PersistentJittedBodyInfo;
+namespace TR { class DefaultCompilationStrategy; }
+namespace TR { class ThresholdCompilationStrategy; }
+namespace TR { class CompilationStrategy; }
+namespace TR { class CompilationInfo; }
+namespace TR { class Options; }
+class TR_J9VMBase;
 class TR_PersistentMethodInfo;
-struct TR_MethodToBeCompiled;
-
-int32_t printTruncatedSignature(char *sigBuf, int32_t bufLen, J9Method *j9method);
+class TR_PersistentJittedBodyInfo;
 
-extern "C" {
-int32_t returnIprofilerState();
-}
+static int32_t interpreterProfilingState = IPROFILING_STATE_OFF;
```

Comment:
This defines a new variable. We want to use the one in HookedByTheJIT.cpp

Reply:
It's mainly for compilation success:
Before - **interpreterProfilingState** was _local_ and _static_ to HookedByTheJIT.cpp, only way to get it is via method returnIprofilerState declared in **CompilationController.hpp**
After - **CompilationController.hpp** removed, new file J9CompilationStrategy.hpp added, in J9CompilationStrategy.cpp there's one place refer the value. 

I can keep using the getter as before if you prefer

Reply:
It's about correctness, not preference. Here's a link that explains the problems with static variables declared in headers: https://stackoverflow.com/questions/3837490/initializing-a-static-variable-in-header